layouts/partials/components/qna_toc.html

```diff
@@ -0,0 +1,17 @@
+{{ $tocJs := resources.Get "js/tableOfContents.js" }}
+
+<script type="text/javascript" src="{{ $tocJs.RelPermalink }}" defer></script>
+
+<div class="card-title pt-6 pb-5">Table of Contents</div>
+<div class="nested-link font-bold py-2">
+  <nav id="TableOfContents">
+    <ul>
+      {{ range $index, $q := . }}
+      <li>
+        <a href="#q{{ $index }}">{{ $q.question }}</a>
+      </li>
+      {{ end }}
+    </ul>
+  </nav>
+</div>
+<div>&nbsp;</div>
```

Comment:
@nonowazu this works fine for the qna template specifically. I honestly have no idea how to go about dynamically building a dict of headers with Hugo in order to generalize this partial beyond qna.